contracts/BalanceLib.sol

```diff
@@ -0,0 +1,146 @@
+/*
+ * Hubii Striim
+ *
+ * Compliant with the Hubii Striim specification v0.12.
+ *
+ * Copyright (C) 2017-2018 Hubii AS
+ */
+
+pragma solidity ^0.4.24;
+
+import {SafeMathInt} from "./SafeMathInt.sol";
+
+library BalanceLib {
+    using SafeMathInt for int256;
+
+    struct Deposit {
+        int256 amount;
+        uint256 timestamp;
+        address token;      //0 for ethers
+        uint256 nonfungible_id;
```

Comment:
As ERC1155 uses ID for both FT and NFT I suggest that properties and params `nonfungible_id` be replaced by `currencyId` (implying also a shift from snake case to camel case), or possibly as simple as `id` as that is used in `BalanceLib`.

In addition as ERC1155 supports multiple tokens per contract I suggest that property `token` be replaced by `contract`.

Reply:
👍